p2p/sender.go

```diff
@@ -223,10 +223,19 @@ func SendReceive(ctx context.Context, tcpNode host.Host, peerID peer.ID,
 		return errors.Wrap(err, "close write", z.Any("protocol", s.Protocol()))
 	}
 
+	zeroResp := proto.Clone(resp)
+
 	if err = reader.ReadMsg(resp); err != nil {
```

Comment:
this will still return the errors that we are seeing in logs:
`Peerinfo failed: read response: peer errored, no response`

how is this allowing zero length responses?

Reply:
See `legacyReadWriter ReadMsg`